plugin-server/src/ingestion/ingestion-consumer.ts

```diff
@@ -588,17 +628,25 @@ export class IngestionConsumer {
                 : this.hub.INGESTION_OVERFLOW_PRESERVE_PARTITION_LOCALITY
 
         await Promise.all(
-            kafkaMessages.map((message) =>
-                this.kafkaOverflowProducer!.produce({
+            kafkaMessages.map((message) => {
+                const headers: MessageHeader[] = message.headers ?? []
+                const existingBreadcrumbs = this.getExistingBreadcrumbsFromHeaders(message)
+                const breadcrumb = this.createBreadcrumb(message)
+                const allBreadcrumbs = [...existingBreadcrumbs, breadcrumb]
```

Comment:
nit: Best to use the concat method on the first array – even though in this case it's unlikely to happen, the spread operator can blow up the call stack, at least with our TypeScript version.